### PR TITLE
Remove all hardcoded yaml files from cardano-testnet

### PR DESCRIPTION
**Pull request by Jimbo4350**
# Description
Hardcoding configuration files makes `cardano-testnet` less portable and also introduces ambiguity into what your node configuration actually is.
# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - roundtrip tests
  - integration tests
  See [Runnings tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Any changes are noted in the `CHANGELOG.md` for affected package
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] Self-reviewed the diff

# Note on CI
If your PR is from a fork, the necessary CI jobs won't trigger automatically for security reasons.
You will need to get someone with write privileges.  Please contact IOG node developers to do this
for you.

### CODE REVIEW

File: cardano-testnet/src/Testnet/Babbage.hs [hunk 1]
```diff
@@ -14,10 +14,12 @@ module Testnet.Babbage
   , babbageTestnet
   ) where
 
+import           Cardano.Api
```
Comment (carbolymer) [inline cardano-testnet/src/Testnet/Babbage.hs:17]: [Import grouping does not adhere to style guide](https://github.com/input-output-hk/cardano-node-wiki/wiki/Style-guide#imports) 🙂

File: cardano-testnet/src/Testnet/Utils.hs [hunk 2]
```diff
@@ -17,20 +22,30 @@ import           Cardano.CLI.Shelley.Output
...
+import qualified Data.ByteString as BS
+import           Data.Text (Text)
 import           Data.Word
 import           GHC.Stack
 import           Options.Applicative
 import           System.Directory (doesFileExist, removeFile)
 
+import           Cardano.Chain.Genesis (GenesisHash (unGenesisHash), readGenesisData)
+
+import qualified Cardano.Crypto.Hash.Blake2b as Crypto
```
Comment (carbolymer) [inline cardano-testnet/src/Testnet/Utils.hs:39]: [Import grouping does not adhere to style guide](https://github.com/input-output-hk/cardano-node-wiki/wiki/Style-guide#imports) 🙂

**Review by carbolymer — APPROVED** [review]

File: cardano-testnet/src/Testnet/Babbage.hs (hunk 1 above)
Comment (Jimbo4350) [reply cardano-testnet/src/Testnet/Babbage.hs]: https://github.com/input-output-hk/cardano-node-wiki/pull/6

File: cardano-testnet/src/Testnet/Utils.hs (hunk 2 above)
Comment (Jimbo4350) [reply cardano-testnet/src/Testnet/Utils.hs:38]: https://github.com/input-output-hk/cardano-node-wiki/pull/6

Comment (Jimbo4350) [reply cardano-testnet/src/Testnet/Utils.hs:38]: Made changes but opened a PR to update the style doc.

**Comment by newhoggy** [comment]
This is good, but I think we should write the default yaml file to a golden file as well so the configuration file can easily be reviewed or read by non-coders.

**Review by newhoggy — APPROVED** [review]

**Comment by Jimbo4350** [comment]
> This is good, but I think we should write the default yaml file to a golden file as well so the configuration file can easily be reviewed or read by non-coders.

Done :+1:

**Pull request merged**